kroxylicious/src/main/java/io/kroxylicious/proxy/frame/BareSaslResponse.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.kroxylicious.proxy.frame;
+
+/**
+ * Ancient versions of Kafka implemented SASL/GSSAPI by sending the response
+ * on the wire as length prefixed bytes (no Kafka protocol header).
+ * This frame represents those kinds of response.
+ */
+public class BareSaslResponse implements ResponseFrame {
```

Comment:
In the sense that `OpaqueRequestFrame` assumes that the frame had a standard Kafka header the answer is no, because the original support for SASL didn't use the `SaslAuthenticate` RPC at all. The client just sent the SASL bytes without any framing. _Maybe_ we could repurpose `OpaqueRequestFrame` to also handle this corner case, but it's not immediately clear to me what compromises that would come with: If we only had to check for the special case in a few places maybe that would be OK, but if many accesses required a check for the special case then I think having a separate class is cleaner. 

This comment does highlight that I've not actually implemented support for this bare request frame in the `KafkaRequestDecoder`. Over all I think I'd prefer to leave decode support as a TODO issue for now. I think the important thing is that the authentication handler is build to handle it.